scripts/pre-push.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+GRAY='\033[30m'  
+RED='\033[31m' 
+YELLOW='\033[33m'
+NC='\033[0m' 
+
+FILES=$(git diff origin/master...HEAD --name-only --diff-filter=d | grep \\.ts)
+
+if [ -n "$FILES" ]
+then
+    echo -e -n $YELLOW
+    echo "------------------------------------------------------"
+    echo -e "Checking .ts files before pushing..."
+    echo "------------------------------------------------------"
+    echo -e -n $GRAY
+    echo "$FILES" | tr ' ' '\n' | xargs -I {} echo "- {}"
+    echo -e $NC
+
+    lint=$(NODE_OPTIONS="--max-old-space-size=8192" npx eslint $FILES | grep -E 'problems')
+    if [ -n "$lint" ]
+    then
+        echo -e "[eslint] ${RED}${lint}${NC}"
+    else
+        echo -e "[eslint] 🚀🚀🚀"
+    fi
+
+    echo "$FILES" | tr ' ' '\n' | xargs -I {} npx prettier {} --check --loglevel warn 
```

Comment:
Can we just auto-run prettier instead of only checking for it? Prettier should cause no failures due to it only being spacing changes.

Reply:
Sure, let's discuss tomorrow in the parking lot... it's just replacing --check with --write I think

Reply:
Kind of redundant for prettier no? We have a pre-commit hook to run prettier (write).

Reply:
don't believe this is a shared DX, I didn't see it in my env until now (in scripts)... husky provides a better DX I think

Reply:
I agree Husky is definitely preferable just saying we should probably re-use the existing `prettier-new-work.sh` script which the `add-prettier-precommit` yarn script sets up. In other words, keep the same end-effect (prettier-write on commit) but move execution to husky and remove `add-prettier-precommit`.

Reply:
yep, makes sense, although it's just a one liner since we already have the git diff files... let's discuss it tomorrow

Reply:
@timolegros New prettier script is easier because husky handles the recursive commit problem I was running into (where the precommit triggered itself).

I think this PR is good to go, could you resolve this conversation if you are fine with the outcome?

Reply:
@timolegros we moved pre-commit to husky with a simpler diff strategy, just git diff head~1 to avoid formatting the entire PR on every commit